### PR TITLE
uWSGI + Web-sockets support

### PR DESCRIPTION
**Pull request by shirishgoyal**
- Redis based sessions
- Replaced gunicorn with uWSGI
- Web-sockets support using Redis
- SocketManager for pushing messages to frontend
- Added angular web-sockets support
- Heroku updates for web-sockets and pgbouncer
- Added test case for broadcast on Task-Feed view (can be tested on [Playground Task-Feed](https://daemo-playground.herokuapp.com/task-feed) with multiple browser tabs open). Removed from code though.
- Updated README

image

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,26 +1,33 @@
...
-google-api-python-client
-gunicorn==19.4.5
-hashids
-httpie
+django-websocket-redis==0.4.6
+dropbox==3.23
+gevent==1.0.2
+google-api-python-client==1.4.1
+hashids==1.1.0
+hiredis==0.2.0
```
Comment (dmorina) [inline requirements.txt]: is this required, considering we have redis package?

**Comment by dmorina** [comment]
running `uwsgi uwsgi-dev.ini` generates this (any requirements?)
realpath() of uwsgi-dev.ini failed: No such file or directory [core/utils.c line 3616]

**Comment by dmorina** [comment]
could you test mac os x

File: requirements.txt (hunk 1 above)
Comment (shirishgoyal) [inline requirements.txt]: Yes for websocket lib as it is a wrapper layer over Redis to handle long live connections.

**Comment by shirishgoyal** [comment]
Did you copy uwsgi-dev-default.ini to uwsgi-dev.ini as mentioned in README ?
Please review README too.

File: requirements.txt (hunk 1 above)
Comment (dmorina) [inline requirements.txt]: okie, i thought it was using https://pypi.python.org/pypi/redis/ 
nvm then :)

Comment (shirishgoyal) [inline requirements.txt]: hiredis can be removed. I thought you referred to django-websocket-redis

**Comment by dmorina** [comment]
looks good to meeeee :D

**Pull request merged**